File: src/main/java/us/illyohs/civilmagiks/client/core/util/FxUtil.java
package us.illyohs.civilmagiks.client.core.util;

import net.minecraft.client.renderer.GlStateManager;
import net.minecraft.client.renderer.Tessellator;
import net.minecraft.client.renderer.VertexBuffer;
import net.minecraft.client.renderer.block.model.ModelResourceLocation;
import net.minecraft.client.renderer.vertex.DefaultVertexFormats;
import net.minecraft.item.Item;
import net.minecraft.tileentity.TileEntity;
import net.minecraft.util.ResourceLocation;
import net.minecraftforge.client.model.ModelLoader;
import net.minecraftforge.fml.client.FMLClientHandler;

import java.awt.*;

public class FxUtil {

    public static void bindBasicItemModel(Item item) {
        ModelLoader.setCustomModelResourceLocation(item, 0, new ModelResourceLocation(item.getRegistryName(), "inventory"));
    }

    public static void bindTexture(ResourceLocation texture) {
        FMLClientHandler.instance().getClient().renderEngine.bindTexture(texture);
    }

    public static void shaderBind(String path) {

    }

}

File: src/main/scala/us/illyohs/civilmagiks/client/util/FxUtil.scala
package us.illyohs.civilmagiks.client.util

import net.minecraft.client.renderer.block.model.ModelResourceLocation
import net.minecraft.item.Item
import net.minecraft.util.ResourceLocation
import net.minecraftforge.client.model.ModelLoader
import net.minecraftforge.fml.client.FMLClientHandler

object FxUtil {
  def bindBasicItemModel(item: Item) {
    ModelLoader.setCustomModelResourceLocation(item, 0, new ModelResourceLocation(item.getRegistryName, "inventory"))
  }

  def bindTexture(texture: ResourceLocation) {
    FMLClientHandler.instance.getClient.renderEngine.bindTexture(texture)
  }

  def shaderBind(path: String) {
  }
}